specs/ServerCertificate.md

```diff
@@ -0,0 +1,289 @@
+             Callback<ICoreWebView2ReceivingServerCertificateErrorEventHandler>(
+             [this](
+               ICoreWebView2* sender,
+               ICoreWebView2ReceivingServerCertificateErrorEventArgs* args) {
+               COREWEBVIEW2_WEB_ERROR_STATUS errorStatus;
+               CHECK_FAILURE(args->get_ErrorStatus(&errorStatus));
+
+               wil::com_ptr<ICoreWebView2Certificate> certificate = nullptr;
+               CHECK_FAILURE(args->get_ServerCertificate(&certificate));
+
+               // Continues with the navigation to a server with a TLS certificate if
+               // the error status is of type `COREWEBVIEW2_WEB_ERROR_STATUS_CERTIFICATE_IS_INVALID`
+               // and trusted by the host app.
+               if (errorStatus == COREWEBVIEW2_WEB_ERROR_STATUS_CERTIFICATE_IS_INVALID &&
+                            ValidateServerCertificate(certificate.get()))
+               {
+                 CHECK_FAILURE(args->put_Handled(TRUE));
+               }
+               else
+               {
+                 // Cancel the request for other TLS certificate error types or if untrusted by the host app.
+                 CHECK_FAILURE(args->put_Cancel(TRUE));
+               }
+               return S_OK;
+             })
+             .Get(),
+         &m_ServerCertificateErrorToken));
+    }
+    else
+    {
+      CHECK_FAILURE(m_webview->remove_ReceivingServerCertificateError(
+        m_ServerCertificateErrorToken));
+      m_ServerCertificateErrorToken.value = 0;
+    }
+  }
+  else
+  {
+    FeatureNotAvailable();
+  }
+}
+
+// Function to validate the server certificate for untrusted root or self-signed certificate.
+bool ValidateServerCertificate(ICoreWebView2Certificate* certificate)
+{
+    // Get the list of host app trusted certificates and its thumbprint.
+
+    // Get last chain element using `get_PemEncodedIssuerCertificateChain` of `ICoreWebView2Certificate`
+    // that contain raw data of untrusted root CA/self signed certificate. Get the untrusted
+    // root CA/self signed certificate thumbprint from the raw certificate data and
+    // validate thumbprint against the host app trusted certificate list.
+
+    // Return true if it exist in host app certificate trusted list, otherwise return false.
+    return true;
+}
+
+// This example clears the TLS decision in response to proceeding with TLS certificate errors.
+if (m_webview)
+{
+  CHECK_FAILURE(m_webView->ClearServerCertificateErrorOverrideCache(
+        Callback<
+            ICoreWebView2ClearServerCertificateErrorOverrideCacheCompletedHandler>(
+            [](HRESULT error, PCWSTR result) -> HRESULT {
+                CHECK_FAILURE(error);
+                if (error != S_OK)
+                {
+                  ShowFailure( error,  L"Clear server certificate error override cache failed");
+                }
+                MessageBox(nullptr, L"Cleared", L"Clear server certificate error override cache", MB_OK);
```

Comment:
We shouldn't be doing message loops inside async callbacks or event handler callbacks. You can use the App::RunAsync method in our sample app to run the message box later not during the callback. There should be some examples if you search for that